src/@ionic-native/plugins/hyper-track/index.ts

```diff
@@ -0,0 +1,347 @@
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with an "OK" string if successful, or it gets rejected if an error ocurred. An "OK" response doesn't necessarily mean that the userId was found. It just means that it was set correctly.
+   *
+   * @see {@link https://docs.hypertrack.com/sdks/cordova/reference.html#methods|Hypertrack Cordova Methods Reference}
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#L20|HyperTrack.js setUserId implementation}
+   *
+   * @usage
+   * ```
+   * HyperTrack.setUserId("userId")
+   *  .then(success => {
+   *    // handle success. Should be "OK".
+   *  }, error => {
+   *    // handle error
+   *  });
+   * ```
+   */
+  @Cordova()
+  setUserId(userId: String): Promise<any> { return; }
+
+  /**
+   * Enable the SDK and start tracking. This will fail if there is no user set.
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with the userId (String) of the User being tracked if successful, or it gets rejected if an error ocurred. One example of an error is not setting a User with getOrCreateUser() or setUserId() before calling this function.
+   *
+   * @see {@link https://docs.hypertrack.com/sdks/cordova/reference.html#methods|Hypertrack Cordova Methods Reference}
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#L27|HyperTrack.js startTracking implementation}
+   *
+   * @usage
+   * ```
+   * HyperTrack.startTracking().then(userId => {
+   *  // handle success
+   * }, error => {
+   *  // handle error
+   * });
+   * ```
+   */
+  @Cordova()
+  startTracking(): Promise<any> { return; }
+
+  /**
+   * Create and assign an action to the current user using specified parameters
+   * All parameters are required, even if some of HyeprTrack's examples say
+   * otherwise, because when left as optional and not passed, it won't work nor
+   * will it raise an error.
+   *
+   * @param {String} [type] The action type. Can be one from "pickup", "delivery", "dropoff", "visit", "stopover" or "task"
+   * @param {String} [lookupId] The Action's desired lookupId
+   * @param {String} [expectedPlaceAddress] The address of the Action
+   * @param {Number} [expectedPlaceLatitude] The latitude of the Action
+   * @param {Number} [expectedPlaceLongitude] The longitude of the Action
+   *
+   * @returns {Promise<any>} Returns a Promise that resolves with a string representation of the Action's JSON, or it gets rejected if an error ocurred.
+   *
+   * @see {@link https://docs.hypertrack.com/sdks/cordova/reference.html#methods|Hypertrack Cordova Methods Reference}
+   * @see {@link https://github.com/hypertrack/hypertrack-cordova/blob/master/www/HyperTrack.js#33|HyperTrack.js createAndAssignAction implementation}
+   * @see {@link https://docs.hypertrack.com/api/entities/action.html#the-action-object|HyperTrack API Documentation for Action}
+   *
+   * @usage
```

Comment:
All usage examples should stay at the top, in the `@Plugin()` decorator's doc section.